cmake/secp256k1.cmake

```diff
@@ -57,4 +57,6 @@ if(MSVC)
   )
 endif()
 
-target_link_libraries(secp256k1 PRIVATE core_interface)
+# Using core_base_interface instead of core_interface
+# to avoid C++-specific flags from warn_interface.
```

Comment:
I think this comment can be improved. We are avoiding passing warning flags to libsecp256k1 because it's a subtree, and warnings produced aren't fixable in our tree, making them pointless, not because it's C code.

Reply:
Thanks! Reworked.